File: Boyer-Moore Majority Vote Algortihm/Python/Boyer-Moore-Majority-Vote-Algortihm.py
def find_majority_element(arr):
    candidate = 0
    count = 0

    for num in arr:
        if count == 0:
            candidate = num
        count += 1 if num == candidate else -1

    return candidate

arr = [2, 2, 1, 1, 2, 2]

result = find_majority_element(arr)
print("Majority element is:", result)

File: Boyer-Moore Majority Vote Algortihm/Go/Boyer-Moore-Majority-Vote-Algortihm.go
package main

import "fmt"

func find_majority_element(arr []int) int {
    candidate := 0
    count := 0

    for _, num := range arr {
        if count == 0 {
            candidate = num
        }
        if num == candidate {
            count++
        } else {
            count--
        }
    }

    return candidate
}

func main() {
    arr := []int{2, 2, 1, 1, 2, 2}

    result := find_majority_element(arr)
    fmt.Println("Majority element is:", result)
}
